src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -1099,16 +967,16 @@ void emitStaticInvoke(MemberName member, Name name) {
         assert(!(member.getDeclaringClass().isInterface() && refKind == REF_invokeVirtual));
 
         // push arguments
-        emitPushArguments(name, 0);
+        emitPushArguments(cob, name, 0);
 
         // invocation
         if (member.isMethod()) {
             mtype = member.getMethodType().toMethodDescriptorString();
-            mv.visitMethodInsn(refKindOpcode(refKind), cname, mname, mtype,
+            cob.invokeInstruction(refKindOpcode(refKind), cname, mname, MethodTypeDesc.ofDescriptor(mtype),
```

Comment:
This can be simplified to define a local variable to create a `MethodTypeDesc`  instead of using `String mtype` and  the else-clause would define a local variable to create a `ClassDesc` instead.   

```
        if (member.isMethod()) {
            var methodTypeDesc = MethodTypeDesc.ofDescriptor(member.getMethodType().descriptorString());
            cob.invokeInstruction(refKindOpcode(refKind), desc, mname, methodTypeDesc,
                                  member.getDeclaringClass().isInterface());
        } else {
            var fieldTypeDesc = ClassDesc.ofDescriptor(member.getFieldType().descriptorString());
            cob.fieldInstruction(refKindOpcode(refKind), desc, mname, fieldTypeDesc);
        }
```

Reply:
fixed, thanks.